Readme.md

```diff
@@ -29,37 +12,44 @@ Firefox, Safari and Edge not supported as they are not yet properly implementing
 -   Edge: <https://developer.microsoft.com/en-us/microsoft-edge/platform/issues/11963735/>
 -   Safari: <https://bugs.webkit.org/show_bug.cgi?id=171934>
 
-## Quick start
-
--   Clone repository, this will be the $GOPATH
-
--   In your editor settings, set $GOPATH to repo if needed
-
--   Install dependencies:
+## Development
 
-        make deps
+### Prerequisites
 
--   Build and run
+[Nix](https://nixos.org/nix) is required for installing dependencies and providing a suitable development environment.
 
-        make
-        ./bin/dividat-driver
+### Quick start
 
-    Or use `go run`:
+- Create a suitable environment: `nix-shell`
+- Build the driver: `make`
+- Run the driver: `./bin/dividat-driver`
 
-        go run src/cmd/dividat-driver/main.go start
-
-## Tests
+### Tests
 
 Run the test suite with: `make test`.
 
-## Releasing
+### Releasing
 
 Currently releases can only be made for Linux (from Linux). Builds on Linux are statically linked with [musl](https://www.musl-libc.org/).
 
 To create a release run: `make release`. You will need to be able to provide appropriate signing keys.
 
 To deploy a new release run: `make deploy`. This can only be done if you are on `master` or `develop` branch, have correctly tagged the revision and have AWS credentials set in your environment.
 
+### Go packages
+
+Go dependencies are provided by the [Go machinery](https://nixos.org/nixpkgs/manual/#sec-language-go) in Nix.
+
+For local development you may use `dep` to install go dependencies: `cd src/dividat-driver && dep ensure`.
+
+New Go dependencies can be added with `dep` (e.g. `dep ensure -add github.com/something/supercool`). The specification of dependencies needs to be recreated for Nix. This can be done with the command `make nix/deps.nix`. Check in the updated `Gopkg.toml`, `Gopkg.lock` and `nix/deps.nix` files.
+
+### Cross compilation
+
+A default environment (defined in `default.nix`) provides all necessary dependencies for building on your native system (i.e. Linux or Darwin). Running `make` will create a binary that should run on your system (at least in the default environemnt).
```

Comment:
This being under the header "cross compilation" is a bit confusing. Let's say very explicitly that crossbuilding is only supported on Linux, _but_ `make` will create a binary for the host system for development.

Reply:
Yop, I merged the section "Cross compilation" with "Releasing" to subsections "Release/Building" and "Release/Deploying".

> I think phoniness is a bit rampant in this Makefile, let's reduce that at least where it leaks into side remarks in the Readme.

True. The root cause of this is that we do not really track root dependencies (i.e. go source files) with the Makefile but instead rely on the go compiler just being fast enough to compile everything (or do internal caching). It is a slightly perverted use of `Make` but I think it is legitimized in our case. Also by the fact that targets are mostly actions (e.g. `build`, `test`, `deploy`) which are really "phony" (i.e. not artifacts on the filesystem). Reducing phonyness requires restructuring the Makefile, which I don't think is absolutely necessary at this moment.

> As we plan to still produce driver versions for all three platforms, also add definitions to build for Mac and Win.

Makefile targets are there and in right format but commented out. I prefer adding them/uncommenting them when we have a working release target system definition.